### PR TITLE
Mayhem integration

### PR DESCRIPTION
**Pull request by code-terror**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -1,17 +0,0 @@
-name: Rust
```
Comment (xansec) [inline .github/workflows/rust.yml]: Please restore upstream workflows.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Build Stage
```
Comment (xansec) [inline Dockerfile:1]: You don't need to copy the Mayhemfile here, since it's parsed from the original clone of the repo.

File: Mayhemfile [hunk 3]
```diff
@@ -0,0 +1,6 @@
+project: code-terror/fast_rsync
```
Comment (xansec) [inline Mayhemfile:1]: I recommend changing the target name so that it is clear what's being fuzzed.

File: README.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ A faster implementation of [librsync](https://github.com/librsync/librsync) in
 pure Rust, using SIMD operations where available. Note that only the legacy MD4
```
Comment (xansec) [inline README.md]: Is this change correct? If so, I recommend making a separate PR directly to upstream rather than changing it here.

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: README.md (hunk 4 above)
Comment (code-terror) [reply README.md]: Sorry for the change..I dont remember the reason..But it is an error

**Comment by code-terror** [comment]
@xansec Pushed the changes..

**Review by xansec — APPROVED** [review]
LGTM!

**Pull request merged**